src/openboardview/FileFormats/XZZPCBFile.cpp

```diff
@@ -0,0 +1,890 @@
+			current_pointer += 1;
+			std::string comment = "";
+			while (buf[current_pointer] != 0x0D) {
+				comment += buf[current_pointer];
+				current_pointer += 1;
+			}
+			comment = read_cb2312_string(comment);
+			diode_dict[net]["0"] = comment; // Seemingly is there is a second reading the first is replaced
+		}
+    }
+}
+
+// Layers:
+// 1->16 Trace Layers (Used in order excluding last which always uses 16)
+// 17 Silkscreen
+// 18->27 Unknown
+// 28 Board edges
+
+void XZZPCBFile::parse_arc_block(std::vector<uint32_t> &buf) {
+	uint32_t layer = buf[0];
+    uint32_t x = buf[1];
+    uint32_t y = buf[2];
+    int32_t r = buf[3];
+    int32_t angle_start = buf[4];
+    int32_t angle_end = buf[5];
+    int32_t scale = buf[6];
+    // int32_t unknown_arc = buf[7];
+	scale = 10000;
+	if (layer != 28) {
+		return;
+	}
+
+	x = static_cast<int>(x) / scale;
+	y = static_cast<int>(y) / scale;
+	r = static_cast<int>(r) / scale;
+	angle_start = static_cast<int>(angle_start) / scale;
+	angle_end = static_cast<int>(angle_end) / scale;
+	BRDPoint centre = {x, y};
+
+	std::vector<std::pair<BRDPoint, BRDPoint>> segments = xzz_arc_to_segments(angle_start, angle_end, r, centre);
+	std::move(segments.begin(), segments.end(), std::back_inserter(outline_segments));
+}
+
+void XZZPCBFile::parse_line_segment_block(std::vector<uint32_t> &buf) {
+    int32_t layer = buf[0];
+    int32_t x1 = buf[1];
+    int32_t y1 = buf[2];
+    int32_t x2 = buf[3];
+    int32_t y2 = buf[4];
+    int32_t scale = buf[5];
+	scale = 10000;
+    int32_t trace_net_index = buf[6];
+	if (layer != 28) {
+		return;
+	}
+	// printf("Line segment block\n");
+	// printf("Layer: %d, x1: %d, y1: %d, x2: %d, y2: %d, scale: %d, trace_net_index: %d\n", layer, x1, y1, x2, y2, scale, trace_net_index);	
+	BRDPoint point;
+	point.x = static_cast<double>(x1) / scale;
+	point.y = static_cast<double>(y1) / scale;
+	BRDPoint point2;
+	point2.x = static_cast<double>(x2) / scale;
+	point2.y = static_cast<double>(y2) / scale;
+	outline_segments.push_back({point, point2});
+}
+
+void XZZPCBFile::parse_part_block(std::vector<char> &buf) {
+	BRDPart blank_part;
+	BRDPin blank_pin;
+	BRDPart part;
+	BRDPin pin;
+
+	des_decrypt(buf);
+
+	uint32_t current_pointer = 0;
+	uint32_t part_size = *reinterpret_cast<uint32_t*>(&buf[current_pointer]);
+	current_pointer += 4;
+	current_pointer += 18;
+	uint32_t part_group_name_size = *reinterpret_cast<uint32_t*>(&buf[current_pointer]);
+	current_pointer += 4;
+	current_pointer += part_group_name_size;
+	
+	// So far 0x06 sub blocks have been first always
+	// Also contains part name so needed before pins
+	if (buf[current_pointer] != 0x06) {
+		exit(1);
+	}
+	
+	current_pointer += 31;
+	uint32_t part_name_size = *reinterpret_cast<uint32_t*>(&buf[current_pointer]);
+	current_pointer += 4;
+	std::string part_name(reinterpret_cast<char*>(&buf[current_pointer]), part_name_size);
+	current_pointer += part_name_size;
+
+	part.name = strdup(part_name.c_str());
```

Comment:
This leaks memory, however we may not have a better way to deal with it right now. Would probably need to convert `BRDPart::name` to an `std::string`.